app/gui/view/component-browser/component-group/src/lib.rs

```diff
@@ -5,6 +5,16 @@
 //!
 //! To learn more about component groups, see the [Component Browser Design
 //! Document](https://github.com/enso-org/design/blob/e6cffec2dd6d16688164f04a4ef0d9dff998c3e7/epics/component-browser/design.md).
+//!
+//! # Header and its Shadow
+//!
+//! To simulate scrolling of the component group entries we move the header of the component group
+//! down while moving the whole component group up (see [`Frp::set_header_pos`]). When the header
+//! is moved down the shadow appears below it. The shadow changes its intensity smoothly before
+//! the header reaches the [`HEADER_SHADOW_PEAK`] distance from the top of the component group.
+//! After that the shadow is unchanged. Near the bottom of the component group we gradually reduce
+//! the size of the shadow so that it will never be rendered outside the component group
```

Comment:
Minor comment: `Near the bottom of the component group` - is IMO not clear enough. IMO this should be clearly written that it happens when we scroll so much that the header reaches the bottom of the component group.

Reply:
Rephrased